File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key
def compare(a, b):
    a1 = int(a + b)
    b1 = int(b + a)
    return b1 - a1


def solution(numbers):
    answer = ''
    strNum = list(map(lambda v:str(v), numbers))
    strNum.sort(key=cmp_to_key(compare))
    return str(int("".join(strNum)))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
class Solution {
    public String solution(int[] numbers) {
        String[] nums= Arrays.stream(numbers).boxed().map(x->x.toString()).toArray(String[]::new);
        List<String> list = new ArrayList<>(Arrays.asList(nums));
        Collections.sort(list, (a,b)->Integer.parseInt(b+a) - Integer.parseInt(a+b));
        
        if(list.get(0).equals("0")) return "0";
        String ans = "";
        for(String str: list){
            ans+=str;
        }
        return ans;
    }
}